packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,294 @@
+  }
+
+  return paginationRange;
+}
+
+function Pagination({
+  className,
+  compact,
+  currentPage,
+  customUrl,
+  onPageChange,
+  leftLabel,
+  rightLabel,
+  totalPages,
+}: PaginationProps): React.ReactElement {
+  const classes = classNames('ds-c-pagination', className);
+
+  /**
+   * `useState` and `useEffect` determine if
+   * mobile layout of component is rendered.
+   */
+
+  const [isMobile, setIsMobile] = React.useState(false);
+  React.useEffect(() => {
+    // Mobile media query derived from: https://design.cms.gov/guidelines/responsive/
+    const media = window.matchMedia('(max-width: 543px)');
+    if (media.matches !== isMobile) {
+      setIsMobile(media.matches);
+    }
+    const listener = () => {
+      setIsMobile(media.matches);
+    };
+
+    media.addEventListener('change', listener);
+    return () => media.removeEventListener('change', listener);
+  }, [isMobile]);
+
+  const pageChange = React.useCallback(
+    (page) => (evt: React.MouseEvent) => onPageChange(evt, page),
+    [onPageChange]
+  );
+
+  const pages = [];
+
+  /**
+   * If `compact` or `isMobile` is true,
+   * don't run code to populate `pages[]`.
+   */
+  if (!compact || !isMobile) {
+    const pageRange = paginationBuilder(currentPage, totalPages);
+
+    if (pageRange[0] >= 2) {
+      /**
+       * If `pageRange` begins with a page of 2 or greater,
+       * begin Pagination with Page 1
+       */
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key="page-1"
+          index={1}
+          isActive={currentPage === 1}
+          onPageChange={pageChange(1)}
+        />
+      );
+
+      /**
+       * If `pageRange` doesn't equal 2, second Pagination element is Ellipses,
+       * otherwise page count continues.
+       */
+      if (pageRange[0] !== 2) {
+        pages.push(<Ellipses key="ellipses-1" />);
+      }
+    }
+
+    /**
+     * Renders all Page components in range (3 pages) to Pagination component.
+     */
+
+    pageRange.map((page) => {
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key={`page-${page}`}
+          index={page}
+          isActive={currentPage === page}
+          onPageChange={pageChange(page)}
+        />
+      );
+    });
+
+    /**
+     * Defines if/when the Ellipses component appears
+     * at the end of the Pagination component -
+     * as long as there are fewer than 7 pages.
+     */
+    if (currentPage <= totalPages - 3 && totalPages > maxVisiblePages) {
+      if (currentPage < totalPages - 3) {
+        pages.push(<Ellipses key="ellipses-2" />);
+      }
+
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key={`page-${totalPages}`}
+          index={totalPages}
+          isActive={currentPage === totalPages}
+          onPageChange={pageChange(totalPages)}
+        />
+      );
+    }
+  }
+
+  return (
+    <nav className={classes} role="navigation" aria-label="Pagination Navigation">
```

Comment:
A few thoughts here:

`aria-label="Pagination Navigation"`: Let's remove "Navigation" since most screen readers will announce the `<nav>` element as navigation by default.

`aria-label="Pagination`: What do you think about making this a prop? We don't know what type of pagination will be created with this component. But sometimes, a more detailed label could be helpful. Like "Plan pagination" for Healthcare.gov or Medicare.gov.

Reply:
One more thought on this part, `role="navigation"`: I'm leaning toward removing this. I know this came up recently in our accessibility channel though. It's good for older browsers/assistive tech. But I know the W3C HTML validator lists it as a warning if you put roles on HTML5 elements that already map to that role.